### ISSUE TITLE
Tuple element is fetched even though discarded

### ISSUE THREAD

**Issue by alrz** [dotnet/roslyn#16166]
**Version Used**:  https://github.com/dotnet/roslyn/commit/a2df65047264ed08fb3a985056085dbcee821253

**Steps to Reproduce**:

```cs
int M()
{
  var (_, e) = (1, 2);
  return e;
}
```
**Expected Behavior**:

(decompiled)
```cs
int M()
{
  var e = (1, 2);
  return e.Item2;
}
```

**Actual Behavior**:

```cs
int M()
{
  var e = (1, 2);
  int item = e.Item1;
  return e.Item2;
}
```

**Comment by alrz** [comment]
I suppose it should be elided even if named but not used?

**Comment by alrz** [comment]
I suppose it should be elided even if named but not used?

**Comment by gafter** [comment]
I wouldn't call it a bug, exactly, but it is an opportunity to produce better code.

**Comment by alrz** [comment]
Note: as a consequence, compiler emits *two* locals instead of *zero* if we discard the second element.

**Comment by Logerfo** [comment]
Would the code below be an an achievable optimization?
```
int M()
{
  return 2;
}
```

**Comment by alrz** [comment]
@Logerfo Yes. Looking at the tests, that's what has been implemented.

**Comment by Logerfo** [comment]
@alrz is it? http://tryroslyn.azurewebsites.net/#b:master/f:r/MYGwhgzhAEDC0G8BQ1oEsB2AXaBZAFAJQqImoBuYATtPgPoA00ApodALy0CMTATIQG4y0KsywBXKhhZDUAXyRygA

**Comment by alrz** [comment]
It's not merged yet. See the linked PR.

**Comment by Logerfo** [comment]
@alrz Oh, I see. That's great, then.

**Comment by jcouv** [comment]
Thanks for reporting this issue.
This optimization was implemented along with a number of other deconstruction optimizations which avoid invoking the ValueTuple constructor. 
Fixed by #17027
